server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldFetcher.java

```diff
@@ -170,6 +219,22 @@ private void collectUnmappedList(Map<String, DocumentField> documentFields, Iter
         }
     }
 
+    public static List<String> getShortestPrefixes(List<String> in) {
+        List<String> shortestPrefixes = new ArrayList<>();
+        if (in.isEmpty() == false) {
+            Collections.sort(in);
+            String lastAddedEntry = in.get(0);
+            shortestPrefixes.add(lastAddedEntry);
+            for (String entry : in) {
```

Comment:
Will this work correctly when nested field names are naturally prefixes of each other? Like `nested` and `nested2`? I wonder if we need to consult the mapping information directly to get the parent nested mappings. Another edge case that comes to mind is if field names contain special characters that could sort before periods `.`.

Reply:
true, I changed an existing test so it shows this shortcoming and changed the lookup for the nested parents out of all available nested mappers.